packages/market-service/src/index.ts

```diff
@@ -1,32 +1,43 @@
 import {
-  GetByMarketCapType,
   HistoryData,
+  MarketCapResult,
+  MarketData,
   MarketDataArgs,
-  MarketDataType,
   PriceHistoryArgs,
   PriceHistoryType
 } from '@shapeshiftoss/types'
-import { GetByMarketCapArgs } from '@shapeshiftoss/types/src'
+import { findAllMarketArgs } from '@shapeshiftoss/types/src'
 
-import { MarketService } from './api'
 import { CoinGeckoMarketService } from './coingecko/coingecko'
 
-export const getDefaultMarketService = (): MarketService => {
-  return new CoinGeckoMarketService()
-}
+const MarketProviders = [new CoinGeckoMarketService()]
 
-export const getByMarketCap: GetByMarketCapType = async (args?: GetByMarketCapArgs) => {
-  return getDefaultMarketService().getByMarketCap(args)
+export const findAll = async (args?: findAllMarketArgs): Promise<MarketCapResult> => {
+  let result: MarketCapResult | null = null
+  for (let i = 0; i < MarketProviders.length && !result; i++) {
```

Comment:
add a comment here about this `!result` fanciness as i missed this initially, but this is great

Reply:
Done.